src/api/cvc4cpp.cpp

```diff
@@ -3062,19 +3155,27 @@ std::vector<Sort> Solver::mkDatatypeSortsInternal(
                                          dtypedecls[i],
                                          i)
         << "a datatype declaration with at least one constructor";
+    CVC4_API_SOLVER_TRY_CATCH_BEGIN;
     datatypes.push_back(dtypedecls[i].getDatatype());
+    CVC4_API_SOLVER_TRY_CATCH_END;
```

Comment:
Maybe better to put `CVC4_API_SOLVER_TRY_CATCH_BEGIN`/`CVC4_API_SOLVER_TRY_CATCH_END` at the beginning and end of the function block (like the other functions in this file).